### PR TITLE
Add codeception tests

### PR DESCRIPTION
**Pull request by nicksellen**
Initial work to add codeception for testing.

### CODE REVIEW

**Comment by nicksellen** [comment]
@koyan I would not suggest to review this yet :) ... but do feel free to have a nose around.

File: .env.codeception [hunk 1]
```diff
@@ -0,0 +1,10 @@
+TEST_SITE_DB_NAME="planet4_dev"
+TEST_SITE_DB_HOST="db"
+TEST_SITE_DB_USER="root"
```
Comment (koyan) [inline .env.codeception]: I would prefer if we put those (db_user and db_pass) as circleCI project variables

**Comment by koyan** [comment]
We cannot merge this at this moment. It fails the tests

**Comment by nicksellen** [comment]
Ok, local dev performance is much better now having moved test deps into `tests/` sub directory. There are now effectively two composer projects in this repo:
* root directory (as it was before)
* `tests/` has it's own `composer.json` just for installing dependencies related to the tests

The issue was doing dependency resolution across all the dependencies including the test ones was incredibly intensive, using up to 2.5gb of memory here. Having it as a seperate composer project means memory usage was peeking at around 625mb (same as before).

Using a `composer.lock` file would dramatically improve install time, but is somewhat incompatible with this approach here (which relies on fetching the latest version from a git branch). The tests dependencies do use a lock file.

I also tried out https://github.com/hirak/prestissimo for parallel composer install which is definitely worth having in a relevant base image, then composer will use it automatically, and I observed it giving a good speed boost.

**Review by 27Bslash6 — APPROVED** [review]

**Review by 27Bslash6 — APPROVED** [review]

**Comment by 27Bslash6** [comment]
Is it time?

**Review by koyan — APPROVED** [review]

**Pull request merged**